controllers/datamovement_controller.go

```diff
@@ -95,18 +95,19 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			return ctrl.Result{}, nil
 		}
 
-		isLustre2Lustre, _ := r.isLustre2Lustre(ctx, dm)
-		teardownFn := map[bool]func(context.Context, *nnfv1alpha1.NnfDataMovement) (*ctrl.Result, error){
-			false: r.teardownRsyncJob,
-			true:  r.teardownLustreJob,
-		}[isLustre2Lustre]
+		teardownFns := []func(context.Context, *nnfv1alpha1.NnfDataMovement) (*ctrl.Result, error){
+			r.teardownRsyncJob,
+			r.teardownLustreJob,
+		}
 
-		result, err := teardownFn(ctx, dm)
-		log.Info("Teardown", "Result", result, "Error", err)
-		if err != nil {
-			return ctrl.Result{}, err
-		} else if result != nil {
-			return *result, nil
+		for _, teardownFn := range teardownFns {
```

Comment:
I don't understand this - why does an rsync teardown need to be called for a luster2luster operation?

Reply:
The parent is also not associated with a specific directive index, so it could be a parent for rsync and L2L at the same time.

Reply:
Are you saying `r.isLustre2Lustre(ctx, dm)` isn't working? I'm missing something.

Reply:
> The parent is also not associated with a specific directive index, so it could be a parent for rsync and L2L at the same time.

nnf-sos includes the directive index in the name using`indexedResourceName(workflow, index),`

Reply:
The parent NnfDataMovement's Spec is empty.  It had Spec.Monitor=true for most of its life, but it never had any clue  to tell me what kind of children might be pointing at it.

Reply:
Here's the parent.
```
apiVersion: nnf.cray.hpe.com/v1alpha1
kind: NnfDataMovement
metadata:
  creationTimestamp: "2022-08-03T15:47:52Z"
  finalizers:
  - dm.cray.hpe.com
  generation: 1
  labels:
    dws.cray.hpe.com/owner.kind: Workflow
    dws.cray.hpe.com/owner.name: zaphod718
    dws.cray.hpe.com/owner.namespace: default
    dws.cray.hpe.com/workflow.name: zaphod718
    dws.cray.hpe.com/workflow.namespace: default
    nnf.cray.hpe.com/directive_index: "0"
  name: zaphod718
  namespace: default
  ownerReferences:
  - apiVersion: dws.cray.hpe.com/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: Workflow
    name: zaphod718
    uid: 8d4c8aab-0719-4d47-ad64-15716a12e2d4
  resourceVersion: "26939958"
  uid: 857294b7-2e56-41c3-94b2-672fdf34def1
spec:
  monitor: true
status:
  conditions:
  - lastTransitionTime: "2022-08-03T15:47:52Z"
    message: Data movement resource starting
    reason: Success
    status: "False"
    type: Starting
  - lastTransitionTime: "2022-08-03T15:47:52Z"
    message: Data movement resource running
    reason: Success
    status: "True"
    type: Running
  message: Data movement resource running
  startTime: "2022-08-03T15:47:52.096156Z"
  state: Running
  status: Success
```

I see that it does have the directive index label.  Maybe I could use that as a clue.

Reply:
That parent is created by startPreRunState().

Reply:
Ohh oh oh - sorry took me awhile.  Yes, there's only one data movement monitor and yes it needs to clean up L2L and/or rsync ops.

We could be smart and if monitor == true => Clean up both, else clean up specific known... but the monitor case would handle the known.. so I think what you have is fine.

Reply:
Unfortunately, Spec.Monitor is false when we teardown the parent.  That happens in startPostRunState().